src/protocols/ssh/ssh.c

```diff
@@ -281,6 +285,7 @@ void* ssh_client_thread(void* data) {
     }
 
     pthread_mutex_init(&ssh_client->term_channel_lock, NULL);
+    pthread_cond_init(&(ssh_client->ssh_credential_cond), NULL);
```

Comment:
I don't think `term_channel_lock` should reused for `ssh_credential_cond`. It's specific to guarding access to the SSH terminal channel.

A pthread condition needs to have its own mutex which guards modification to the value that is changed when the condition is signaled.